tests/server/test_shutdown.py

```diff
@@ -0,0 +1,19 @@
+import pytest
+
+from utils import assert_msg
+from aio_utils import create_connection
+
+pytestmark = [pytest.mark.functional, pytest.mark.flaky]
+
+
+@pytest.mark.asyncio
+async def test_shutdown(base_url, http_client, server_port):
+    await create_connection(base_url, http_client)
+
+    print("checkpoint 1")
+
+    url = f"ws://127.0.0.1:{server_port}/api/shutdown/"
+    async with http_client.delete(url) as resp:
+        print("checkpoint 2")
+        assert resp.status == 200
+        assert_msg(await resp.json(), "SERVER_SHUTDOWN")
```

Comment:
There are two parts to [the errors](https://pastebin.com/raw/0ELwk8E6) you are getting:

1) the "ERROR at teardown of test_shutdown": "RuntimeError: cannot schedule new futures after shutdown" - this happening because of the `server_port` fixture - that fixture starts a libertem server in the background, which is stopped again after usage. But as you are already shutting down the server, the second try fails when it tries to `close` the executor.

2) the `http_client`, which is based on `aiohttp`, fails to read the response with a `aiohttp.client_exceptions.ServerDisconnectedError`


I think 1) can be fixed by not stopping the server in [the fixture](https://github.com/LiberTEM/LiberTEM/blob/048c38d0cce687dc16269ff71987ec3e7cfafdba/tests/conftest.py#L66). This can be done even without modifying the fixture code - as you can see, the executor is only closed if there is one in the corresponding state variable - you could set it to `None` in your `ShutdownHandler`.

I'm not sure yet how to fix 2) - maybe in addition to `flush`, you need to call [`finish`](https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.finish)? I'm not sure this is enough.